test/Interop/Cxx/destructor/basic.swift

```diff
@@ -0,0 +1,61 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import Basic
+import StdlibUnittest
+
+var CXXDestructorTestSuite = TestSuite("CXX Destructor")
+
+@inline(never)
+func createTypeWithUserProvidedDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+}
+
+@inline(never)
+func createTypeWithEmptyDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let empty = EmptyDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialImplicitDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let implicit = HasNonTrivialImplicitDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialDefaultDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let def = HasNonTrivialDefaultDestructor(box: box)
+}
+
+CXXDestructorTestSuite.test("Basic object with destructor") {
+  let ptr = UnsafeMutablePointer<Int32>.allocate(capacity: 1)
```

Comment:
Could you deallocate `ptr`? Alternatively, make a local variable and use `withUnsafeMutablePointer`.